roles/lib_openshift/library/oc_obj.py

```diff
@@ -1533,14 +1533,11 @@ def run_ansible(params, check_mode=False):
         if state == 'list':
             return {'changed': False, 'results': api_rval, 'state': 'list'}
 
-        if not params['name']:
-            return {'failed': True, 'msg': 'Please specify a name when state is absent|present.'}  # noqa: E501
-
         ########
         # Delete
         ########
         if state == 'absent':
-            if not Utils.exists(api_rval['results'], params['name']):
+            if not api_rval['results'] or not api_rval['results'][0]:
```

Comment:
Best I can tell, as the return of `get()` the value of `api_rval['results']` should always be a list.